Library/Homebrew/os/linux/elf.rb

```diff
@@ -126,8 +127,21 @@ def initialize(path)
 
     private
 
+    def default_on_exception(val, exception)
+      yield
+    rescue exception
+      val
+    end
+
     def needed_libraries(path)
-      if DevelopmentTools.locate "readelf"
+      # adding support for HOMEBREW_PATCHELF_RB here, should be a simple conditional.
+      if HOMEBREW_READELF_RB
+        patcher = path.patcher
+        patch_err = PatchELF::PatchError
```

Comment:
Nit: Remove this variable and use `PatchELF::PatchError` directly below. It's a little clearer what's going on, that way.